services/datasets/datasets/ports/postgres.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+import typing
+
+from sqlalchemy import Table, Column, create_engine, MetaData
+from sqlalchemy import (
+    String,
+)
+
+from dataservices.amsterdam_schema import Dataclass
+
+
+JSON_TYPE_TO_PG = {
+    "string": String
+}
+
+@dataclass
+class DatasetTable:
+    name: str
+    schema: str
+    columns: typing.List[Column]
+
+    @classmethod
+    def from_dataclass(cls, dataset_id: str, dclass: Dataclass):
+        columns = [
+            Column(
+                field.name,
+                JSON_TYPE_TO_PG[field.type]
+            ) for field in dclass.fields
+        ]
+        return cls(
+            name=dclass.name,
+            schema=dataset_id,
+            columns=columns
+        )
+
+    @property
+    def pg_table(self) -> Table:
+        return Table(
+            self.name, MetaData(),
+            schema=self.schema,
+            *self.columns
+        )
+
+
+class PostgresPort:
+    def __init__(self, db_uri: str):
+        self.engine = create_engine(
```

Comment:
De PostgresPort wordt per request geinstantieerd. Maar, het idee achter sqlachemy is dat de engine instance maar een keer wordt geinstantieerd bij start van een proces.

Reply:
fixed @jjmurre